assets/blocks/course-outline/course-block/edit.js

```diff
@@ -12,12 +12,31 @@ import {
 	withColorSettings,
 	withDefaultBlockStyle,
 } from '../../../shared/blocks/settings';
+import { COURSE_STATUS_STORE } from '../status-store';
 
 /**
  * A React context which contains the attributes and the setAttributes callback of the Outline block.
  */
 export const OutlineAttributesContext = createContext();
 
+/**
+ * A hook to update the status store when a lesson is added or removed.
+ *
+ * @param {string} clientId The outline block id.
+ */
+const useUpdateLessonCount = function ( clientId ) {
+	const lessonCount = useSelect( ( selectCount ) => {
+		return selectCount( 'core/block-editor' ).getGlobalBlockCount(
```

Comment:
So currently the way this works is that when the user creates/removes lesson blocks, we keep the changes in block attributes and we only store them in the db when the user clicks update.

Since we only store the lessons when clicking update, there are two different counts for lessons:
1. The lessons that we actually have in the database and are linked to the course.
2. The lessons which are displayed which includes the ones in 1. and the lessons that were just created and are going to be saved in the next update.

If we fetch only the ones included in 1, then the bar would work only for saved lessons and there would be many incosistencies. This is also the reason that we can't use the db ids and we have to go with block ids, as we won't be able to track not stored lessons.

There is also another purpose on using this hook, it's that a rerender is triggered when the number of the lessons changes.

Reply:
> 
    1. The lessons that we actually have in the database and are linked to the course.
    2. The lessons which are displayed which includes the ones in 1. and the lessons that were just created and are going to be saved in the next update.

    ...

    There is also another purpose on using this hook, it's that a rerender is triggered when the number of the lessons changes.

Ok, that makes sense. I still think that using the actual Lesson IDs and making a way to represent not-yet-saved lessons seems safer. But if that, plus the rerender trigger, seems like too much complexity for now then I don't feel super strongly about this. I just don't like the assumption that Lesson Blocks and actual Lessons have a 1:1 relationship, because that could change in the future. Might be a debt that we have to pay off later 🙂